test/loadTest/cache_refresh/reqs_x_secs.sh

```diff
@@ -21,33 +21,68 @@
 # iot_support at tid dot es
 # author: 'Iv�n Arias Le�n (ivan dot ariasleon at telefonica dot com)'
 
-if [  "$1" == ""  ]
-  then
-    echo "ERROR - No url defined (Mandatory)"
-    echo "usage:"
-    echo "    ./reqs_x_secs.sh <url> [-reset]"
-    echo "    example: ./reqs_x_secs.sh localhost:4567 "
-    echo ""
-    echo " the -reset parameter is used to reset the listener previously."
-    echo ""
-    echo "           ( use [ Ctrl+C ] to stop )"
-    exit
-fi
 
-if [  "$2" == "-reset"  ]
-  then
-     curl -s $1/reset > /dev/null
-     echo ""
-     echo " WARN - The listener has been reset... "
+# variables
+LISTENER=localhost:8090
+CB=None
+RESET=false
+notifQueueSize=N/A
+
+# ========================================================
+#
+# usage - show the script usage
+#
+function usage(){
+     echo " "
+     echo "Usage: "$(basename $0)
+     echo "   -u (usage) [OPTIONAL]"
+     echo "   -reset (this parameter is used to reset the listener previously) [OPTIONAL]"
+     echo "   --listener <listener endpoint> (listener endpoint host:port)[OPTIONAL] (default: localhost:8090)"
+     echo "   --cb <CB endpoint> (listener endpoint host:port) [OPTIONAL] (default: None)"
+     echo " "
+     echo " Example:"
+     echo "     "$(basename $0)" --listener localhost:8090 --cb localhost:8090 -reset"
+     echo " "
+     echo " Note: "
+     echo "     - if cb parameter is not used the queue size is not displayed..."
      echo ""
+     echo "                      ( use [ Ctrl+C ] to stop )"
+     echo " "
+     exit 0
+}
+
+# BEGIN
+while [ "$#" != 0 ]
+do
+  if   [ "$1" == "-u" ];          then usage;
+  elif [ "$1" == "--listener" ];  then LISTENER="$2"; shift;
+  elif [ "$1" == "--cb" ];        then CB="$2"; shift;
+  elif [ "$1" == "-reset" ];      then RESET="true"; shift;
```

Comment:
This `shift` will eat a parameter.
Not what you want.

[ it works *only* if -reset is the *last* parameter ]